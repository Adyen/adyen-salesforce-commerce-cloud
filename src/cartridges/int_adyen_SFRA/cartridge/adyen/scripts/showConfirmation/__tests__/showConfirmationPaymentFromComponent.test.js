/* eslint-disable global-require */
let showConfirmationPaymentFromComponent;
let res;
let req;

beforeEach(() => {
  const { adyen } = require('../../../../controllers/middlewares/index');
  showConfirmationPaymentFromComponent =
    adyen.showConfirmationPaymentFromComponent;
  jest.clearAllMocks();
  const adyenConfigs = require('*/cartridge/adyen/utils/adyenConfigs');
  adyenConfigs.getKlarnaInlineWidgetEnabled = jest.fn().mockReturnValue(false);
  res = { redirect: jest.fn(), json: jest.fn() };
  req = {
    form: {
      additionalDetailsHidden: JSON.stringify({
        paymentData: 'mocked_paymentData',
        details: 'mocked_details',
        paymentMethod: {
          type: 'mocked_type'
        }
      }),
      result: null
    },
    locale: { id: 'nl_NL' },
  };
});

afterEach(() => {
  jest.resetModules();
});

describe('Show Confirmation Payment From Component', () => {
  test.each(['Authorised', 'Pending', 'Received'])(
    'should handle successful payment: %p',
    (a) => {
      const adyenCheckout = require('*/cartridge/adyen/scripts/payments/adyenCheckout');
      const URLUtils = require('dw/web/URLUtils');

      adyenCheckout.doPaymentsDetailsCall.mockImplementation(() => ({
        resultCode: a,
      }));
      showConfirmationPaymentFromComponent(req, res, jest.fn());
      expect(URLUtils.url.mock.calls[0][0]).toBe('Order-Confirm');
    },
  );
  it('should redirect on unsuccessful payment', () => {
    const adyenCheckout = require('*/cartridge/adyen/scripts/payments/adyenCheckout');
    const URLUtils = require('dw/web/URLUtils');
    adyenCheckout.doPaymentsDetailsCall.mockImplementation(() => ({
      resultCode: 'Not_Authorised',
    }));
    showConfirmationPaymentFromComponent(req, res, jest.fn());
    expect(URLUtils.url.mock.calls).toMatchSnapshot();
  });
});
