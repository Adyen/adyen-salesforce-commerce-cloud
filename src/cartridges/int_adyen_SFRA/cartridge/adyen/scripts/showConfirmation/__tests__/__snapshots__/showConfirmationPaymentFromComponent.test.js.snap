// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Confirmation Payment From Component should redirect on placeOrder error 1`] = `
[
  [
    "Checkout-Begin",
    "stage",
    "payment",
    "paymentError",
    "mocked_error.payment.not.valid",
  ],
]
`;

exports[`Show Confirmation Payment From Component should redirect on unsuccessful payment 1`] = `
[
  [
    "Checkout-Begin",
    "stage",
    "payment",
    "paymentError",
    "mocked_error.payment.not.valid",
  ],
]
`;
